libqtile/command/base.py

```diff
@@ -54,14 +56,85 @@ class CommandException(Exception):
     """Error raised while executing a command"""
 
 
-class CommandObject(metaclass=abc.ABCMeta):
+class _CommandMeta(abc.ABCMeta):
+    def __new__(cls, name, bases, dct):
+        instance = super().__new__(cls, name, bases, dct)
+
+        if hasattr(cls, "_commands"):
```

Comment:
This should probably be `instance` not `cls`.

Reply:
However, this cause a problem because it sets `CommandObject._commands` which means every object that inherits `CommandObject` will trigger this return.

Do we need to keep a dict of names that have had commands added?